pika/adapters/tornado_connection.py

```diff
@@ -72,7 +73,10 @@ def _adapter_disconnect(self):
         """Disconnect from the RabbitMQ broker"""
         if self.socket:
             self.ioloop.remove_handler(self.socket.fileno())
-        super(TornadoConnection, self)._adapter_disconnect()
+        try:
+            super(TornadoConnection, self)._adapter_disconnect()
+        except exceptions.AMQPConnectionError as e:
```

Comment:
I can see how an exception here is a problem for frameworks like Tornado and Twisted that rely on continuation-passing style. I am not aware of an ideal solution for this in the existing pika code. Does TornadoConnection's `stop_ioloop_on_close=True` cause the ioloop to stop in the #525 scenario? I see that there are other types of exceptions that would be problematic for these frameworks, such as exceptions.ProtocolVersionMismatch.